src/NovelRT/CMakeLists.txt

```diff
@@ -89,7 +89,7 @@ set(SOURCES
 
 add_library(Engine SHARED ${SOURCES})
 
-set_property(TARGET Engine PROPERTY OUTPUT_NAME "NovelRT")
+set_property(TARGET Engine PROPERTY OUTPUT_NAME "NovelRT-$<CONFIG>")
```

Comment:
What's this change do exactly?

Reply:
It allows the artifacts to be merged so that downloading NovelRT allows CMake to choose debug/release builds based on the config.

Reply:
Specifically, it'll generate outputs looking something like this, after having its outputs merged: 
![Screenshot_20220418_150149](https://user-images.githubusercontent.com/21000186/163819223-7fd0492f-b92a-476c-a605-70bbebcf6cc7.png)

Reply:
So this name will carry over to actual released games?

Reply:
I should also clarify from a C API if the name of the DLL is constantly changing we can't do bindings.

Reply:
> So this name will carry over to actual released games?

We could likely drop the prefix for MinSizeRel builds - it was just easier to keep it to simplify the logic.

> I should also clarify from a C API perspective if the name of the DLL is constantly changing we can't do bindings.

For .NET there's [SetDllImportResolver](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.nativelibrary.setdllimportresolver?view=net-6.0#system-runtime-interopservices-nativelibrary-setdllimportresolver(system-reflection-assembly-system-runtime-interopservices-dllimportresolver)) which should handle this just fine.

Reply:
How do we use this via ClangSharp?

Reply:
In addition, how does a game use this?

Reply:
> How do we use this via ClangSharp?

The same way as normal, plus a call to SetDllImportResolver if necessary.

> In addition, how does a game use this?

The same way as normal.

Reply:
So we can call this from ClangSharp's command line? because that's the normal way

Could we set up this logic as part of the C# .NET template when creating a C# NovelRT game?

Reply:
ClangSharp doesn't care about the library - SetDllImportResolver handles that. You don't need to set it up as part of the .NET template because the bindings library will handle calling SetDllImportResolver.

Reply:
Oh OK I think I get you. Thanks!

Reply:
cc @jeuxjeux20 how problematic will this be for Java/Kotlin?

Reply:
> cc @jeuxjeux20 how problematic will this be for Java/Kotlin?

It's actually fine, we can easily choose which DLL to load in Kotlin with `System.load("<find_the_dll>")`. However I wonder how this will affect the "extension" library `NovelRT.JavaSupport.dll` which has a dependency on `NovelRT.dll`

Reply:
Ah right because JNI is AOT C/C++ stuff right?

Reply:
As long as it's built against the right version, it should work just fine.